app/src/view/DatePicker.js

```diff
@@ -0,0 +1,161 @@
+/**
+ * Date picker, wraps ReactDatePicker for now
+ * @author scjody
+ * @since May 2020
+ * @flow
+ */
+
+import React from 'react';
+import { createPortal } from 'react-dom';
+import ReactDatePicker from 'react-datepicker';
+import { makeStyles } from '@material-ui/core/styles';
+import { formatISO, parseISO } from 'date-fns';
+
+import 'react-datepicker/dist/react-datepicker.css';
+
+type FormTarget = { target: { name: string, value: string }};
+type P = {
+  name: string,
+  onChange: FormTarget => void,
+  selected: string,
+  label: string,
+};
+
+const useStyles = makeStyles((theme) => {
+  const borderColor = theme.palette.type === 'light' ? 'rgba(0, 0, 0, 0.23)' : 'rgba(255, 255, 255, 0.23)';
+
+  return {
+    label: {
+      position: 'relative',
+      display: 'inline-block',
+      paddingTop: '6px',
+      marginTop: '16px',
+      marginBottom: '8px',
+      width: '100%',
+      '& .react-datepicker-wrapper': {
+        display: 'inline',
+      },
+    },
+    datepicker: {
+      backgroundColor: theme.palette.background.paper,
+      borderColor: borderColor,
+      borderTopColor: 'transparent',
+      border: 'solid 1px',
+      borderRadius: theme.shape.borderRadius,
+      padding: '18.5px 14px',
+      color: theme.palette.text.primary,
+      fontSize: theme.typography.fontSize,
+      lineHeight: '18px',
+      width: '100%',
+      '&:hover': {
+        borderColor: theme.palette.text.primary,
+        borderTopColor: 'transparent',
+      },
+    },
+    span: {
+      position: 'absolute',
+      top: '0',
+      left: '0',
+      display: 'flex',
+      width: '100%',
+      fontSize: '10px',
+      lineHeight: '15px',
+      '&:before,&:after': {
+        content: '""',
+        display: 'block',
+        boxSizing: 'border-box',
+        marginTop: '6px',
+        borderTop: 'solid 1px',
+        borderColor: borderColor,
+        minWidth: '10px',
+        height: '8px',
+      },
+      '&:hover::before,&:hover::after': {
+        borderColor: theme.palette.text.primary,
+      },
+      '&:before': {
+        marginRight: '4px',
+        borderLeft: 'solid 1px transparent',
+        borderRadius: '4px 0',
+      },
+      '&:after': {
+        flexGrow: '1',
+        marginLeft: '4px',
+        borderRight: 'solid 1px transparent',
+        borderRadius: '0 4px 4px 0',
+        height: '57px',
+      },
+    },
+    '@global': {
+      '#datepickers .react-datepicker__header, #datepickers .react-datepicker__current-month, #datepickers .react-datepicker__day-name, #datepickers .react-datepicker-time__header': {
```

Comment:
I added an id of `#datepickers` to the div that holds the actual pickers to force my selectors to be more specific than the react-datepicker CSS (which doesn't use an ID). An alternative would be to make my selectors as specific as theirs and depend on the fact that later selectors override earlier ones with equal specificity, but those look like `.react-datepicker__time-container .react-datepicker__time .react-datepicker__time-box ul.react-datepicker__time-list li.react-datepicker__time-list-item--selected` (yes, that's _one_ selector) so I think this is less bad.

Reply:
This'll work, so long as we never want multiple datepickers on the page at once. You can also just double up the selector to make it more specific. Ie, Instead of `#datepickers .react-datepicker__header` you can use `.react-datepicker__header.react-datepicker__header`

Reply:
That's an interesting trick, thanks for letting me know!

In this case though we'd need to repeat the selector 6 times to be more specific in some cases (since the declaration we're overriding sometimes has 5 class selectors). I just tried this and one such case needs to be `.react-datepicker__time-list-item.react-datepicker__time-list-item.react-datepicker__time-list-item.react-datepicker__time-list-item.react-datepicker__time-list-item.react-datepicker__time-list-item`. I can make this change if you want (and not _all_ declarations will need 6 selectors) but I think that using the ID selector is cleaner.

If we ever need multiple datepickers on the same page with different styling we can always give the second set of datepickers a different ID. The ID is applied to the portal `<div>` and there's nothing stopping us from having as many of those as we need.